src/App.js

```diff
@@ -1,60 +1,99 @@
-import React from "react";
-import { onChildAdded, push, ref, set } from "firebase/database";
-import { database } from "./firebase";
+import React, { useState, useEffect } from "react";
+import { onChildAdded, push, ref as databaseRef, set } from "firebase/database";
+import { database, storage } from "./firebase";
+import {
+  getDownloadURL,
+  ref as storageRef,
+  uploadBytes,
+} from "firebase/storage";
 import logo from "./logo.png";
 import "./App.css";
+import Card from "react-bootstrap/Card";
 
 // Save the Firebase message folder name as a constant to avoid bugs due to misspelling
-const MESSAGE_FOLDER_NAME = "messages";
-
-class App extends React.Component {
-  constructor(props) {
-    super(props);
-    // Initialise empty messages array in state to keep local state in sync with Firebase
-    // When Firebase changes, update local state, which will update local UI
-    this.state = {
-      messages: [],
-    };
-  }
-
-  componentDidMount() {
-    const messagesRef = ref(database, MESSAGE_FOLDER_NAME);
+const POSTS_FOLDER_NAME = "posts";
+const IMAGES_FOLDER_NAME = "images";
```

Comment:
I know the sample code calls these "FOLDER" but I would rather name these 
```suggestion
const POSTS_KEY = "posts";
const IMAGES_KEY = "images";
```

since these represent the reference to a key in our key/value database

Reply:
Ah sorry this should only be a valid comment for the POSTS_KEY as IMAGES is in firebase storage and actually uses folders.